src/frontend/progress_bar.rs

```diff
@@ -0,0 +1,125 @@
+use gtk::prelude::*;
+use relm4::prelude::*;
+use std::cell::RefCell;
+use std::f64::consts::PI;
+use std::rc::Rc;
+
+pub(crate) const DEFAULT_TOOLTIP_ETA_PROGRESS_BAR: &str = "ETA for next reward payment";
+
+#[derive(Debug, Clone)]
+pub struct CircularProgressBar {
+    progress: Rc<RefCell<f64>>,
+    tooltip_text: String,
+    diameter: f64,
+}
+
+#[derive(Debug)]
+pub enum ProgressBarInput {
+    SetProgress(f64),
+    SetTooltip(String),
+}
+
+#[relm4::component(pub)]
+impl Component for CircularProgressBar {
+    type Init = f64; // Diameter of the progress bar
+    type Input = ProgressBarInput;
+    type Output = ();
+    type CommandOutput = ();
+
+    view! {
+        #[root]
+        gtk::DrawingArea {
+            set_content_width: (model.diameter + 1.0) as i32,
+            set_content_height: (model.diameter + 1.0) as i32,
+            set_margin_top: 10,
+            set_margin_bottom: 10,
+            set_margin_start: 10,
+            set_margin_end: 10,
+            set_visible: true,
+            set_draw_func: {
+                let progress_clone = model.progress.clone();
```

Comment:
No need to add `_clone` suffix to variables:
```suggestion
                let progress = model.progress.clone();
```